bam_sort.c

```diff
@@ -126,26 +126,30 @@ typedef struct {
     bam1_p b;
 } heap1_t;
 
-#define __pos_cmp(a, b) ((a).pos > (b).pos || ((a).pos == (b).pos && ((a).i > (b).i || ((a).i == (b).i && (a).idx > (b).idx))))
-
-static inline int bam1_lt_by_tag(const bam1_p a, const bam1_p b);
+static inline int bam1_cmp_by_tag(const bam1_p a, const bam1_p b);
 
 // Function to compare reads in the heap and determine which one is < the other
 static inline int heap_lt(const heap1_t a, const heap1_t b)
 {
     if (g_is_by_tag) {
         int t;
         if (a.b.b == NULL || b.b.b == NULL) return a.b.b == NULL? 1 : 0;
```

Comment:
Could we have a bit more verbose variable names? I can see that neither `heap_t` nor `bam1_p` are exported, so maybe we can make clearer what `b` stands for in each case.
Also, `bam1_p` is defined in **bam_rmdup.c** as 
`typedef bam1_t *bam1_p;`